incubator/virtualcluster/pkg/syncer/mccontroller/mccontroller.go

```diff
@@ -390,33 +431,43 @@ func (c *MultiClusterController) processNextWorkItem() bool {
 
 	// RunInformersAndControllers the syncHandler, passing it the cluster/namespace/Name
 	// string of the resource to be synced.
-	if result, err := c.Reconciler.Reconcile(req); err != nil {
-		if c.Queue.NumRequeues(obj) >= constants.MaxReconcileRetryAttempts {
-			metrics.RecordDWSOperationStatus(c.objectKind, req.ClusterName, constants.StatusCodeExceedMaxRetryAttempts)
-			c.Queue.Forget(obj)
-			klog.Warningf("%s dws request is dropped due to reaching max retry limit: %+v", c.name, obj)
-			return true
+	result, err := c.Reconciler.Reconcile(req)
+	if err == nil {
+		metrics.RecordDWSOperationStatus(c.objectKind, req.ClusterName, constants.StatusCodeOK)
+		if result.RequeueAfter > 0 {
+			c.Queue.AddAfter(req, result.RequeueAfter)
+		} else if result.Requeue {
+			c.Queue.AddRateLimited(req)
 		}
-		metrics.RecordDWSOperationStatus(c.objectKind, req.ClusterName, constants.StatusCodeError)
-		c.Queue.AddRateLimited(req)
-		klog.Error(err)
-		return false
-	} else if result.RequeueAfter > 0 {
-		c.Queue.AddAfter(req, result.RequeueAfter)
-		return true
-	} else if result.Requeue {
-		c.Queue.AddRateLimited(req)
+		// if no error occurs we Forget this item so it does not
+		// get queued again until another change happens.
+		c.Queue.Forget(obj)
 		return true
 	}
 
-	metrics.RecordDWSOperationStatus(c.objectKind, req.ClusterName, constants.StatusCodeOK)
+	// rejected by apiserver(maybe rejected by webhook or other admission plugins)
+	// we take a negative attitude on this situation and fail fast.
+	if apierr, ok := err.(apierrors.APIStatus); ok {
+		if code := apierr.Status().Code; code == http.StatusBadRequest || code == http.StatusForbidden {
```

Comment:
we cannot assume webhook error filled the `Reason` field. check code is the most reliable solution